### PR TITLE
updated-sdk-example

### PR DESCRIPTION
**Pull request by cedrabner**

### CODE REVIEW

File: docs/build-on-linea/linea-sdk/index.md [hunk 1]
```diff
@@ -27,8 +27,44 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
...
  5. Claiming messages (use one of the get message methods to grab all the parameters values)
 
+ 
+:::info[New Features]
 
-The code example below demonstrates how to initialize the SDK with the relevant configuration options, and how to call the featured methods.
+ The updated Linea SDK package enhances the L1 contract interaction and adds support for the new L1 claiming system, which is based on a merkle tree.
+ 
+ There are two important things to note:
+  - The previous L1 claiming and all function associated are still supported.
```
Comment (thedarkjester) [inline docs/build-on-linea/linea-sdk/index.md:36]: `functions` vs. `function`

File: docs/build-on-linea/linea-sdk/index.md [hunk 2]
```diff
@@ -27,8 +27,44 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
 
  5. Claiming messages (use one of the get message methods to grab all the parameters values)
 
+ 
+:::info[New Features]
 
-The code example below demonstrates how to initialize the SDK with the relevant configuration options, and how to call the featured methods.
+ The updated Linea SDK package enhances the L1 contract interaction and adds support for the new L1 claiming system, which is based on a merkle tree.
```
Comment (thedarkjester) [inline docs/build-on-linea/linea-sdk/index.md:33]: based on a `Merkle` tree `and requires a Merkle proof for claiming`

File: docs/build-on-linea/linea-sdk/index.md [hunk 3]
```diff
@@ -27,8 +27,44 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
...
+ 
+ There are two important things to note:
```
Comment (thedarkjester) [inline docs/build-on-linea/linea-sdk/index.md:35]: An additional point can be made here as well (so there are "three important things to note")

The Previous L1 claiming function and code samples provided here caters for the transition period where pre-transition messages are claimed without the Merkle proof and post-transition with proof. If this SDK is being consumed after the transition, using the logic that switches between Merkle and non-Merkle proof claiming is sub-optimal.

**Review by thedarkjester — CHANGES_REQUESTED** [review]
General comment `merkle` should be `Merkle`

File: docs/build-on-linea/linea-sdk/index.md [hunk 4]
```diff
@@ -27,8 +27,44 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
...
+
+ A new L1ClaimingService class that includes the following functions:
+  - getMessageProof: This function retrieves the message merkle tree proof required for new message claims on L1.
+  - isClaimingNeedingProof: This function determines whether a proof is needed to claim a message.
+  - getMessageStatus: This function retrieves a message's status, returning the status of both old and new messages.
+  - estimateClaimMessageGas: This function provides an estimate of the gas cost for both old and new claim transactions.
+  - claimMessage: This function enables a message to be claimed using either the old or new method.
+
+ Two new functions in the L1 contract:
+  - estimateClaimWithProofGas to estimates the gas cost for new claim transactions.
```
Comment (thedarkjester) [inline docs/build-on-linea/linea-sdk/index.md:49]: `estimate` vs. `estimates`

Comment (thedarkjester) [reply docs/build-on-linea/linea-sdk/index.md:49]: The formatting is generally inconsistent.

e.g. ``- getMessageProof:` has a colon, some places don't like in this one.

Some lines have periods. Some don't.

Some places we call it `functions` and some places `methods` - probably best to align on one.

File: docs/build-on-linea/linea-sdk/index.md (hunk 2 above)
Comment (cedrabner) [reply docs/build-on-linea/linea-sdk/index.md]: fixed

File: docs/build-on-linea/linea-sdk/index.md (hunk 3 above)
Comment (cedrabner) [reply docs/build-on-linea/linea-sdk/index.md]: added

File: docs/build-on-linea/linea-sdk/index.md (hunk 4 above)
Comment (cedrabner) [reply docs/build-on-linea/linea-sdk/index.md]: fixed

File: docs/build-on-linea/linea-sdk/index.md [hunk 5]
```diff
@@ -30,28 +30,29 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
...
   - getMessageStatus: This function retrieves a message's status, returning the status of both old and new messages.
   - estimateClaimMessageGas: This function provides an estimate of the gas cost for both old and new claim transactions.
-  - claimMessage: This function enables a message to be claimed using either the old or new method.
+  - claimMessage: This function enables a message to be claimed using either the old or new function.
 
  Two new functions in the L1 contract:
-  - estimateClaimWithProofGas to estimates the gas cost for new claim transactions.
-  - claimWithProof to claim a message using the new claimMessageWithProof function
+  - estimateClaimWithProofGas: This functions estimates the gas cost for new claim transactions.
+  - claimWithProof: This functions claims a message using the new claimMessageWithProof function.
```
Comment (thedarkjester) [inline docs/build-on-linea/linea-sdk/index.md:51]: this function vs this functions

**Review by Ludcour — APPROVED** [review]

File: docs/build-on-linea/linea-sdk/index.md [hunk 6]
```diff
@@ -27,8 +27,45 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
 
  5. Claiming messages (use one of the get message methods to grab all the parameters values)
 
-
-The code example below demonstrates how to initialize the SDK with the relevant configuration options, and how to call the featured methods.
+ 
+:::info[New Features]
+
+ The updated Linea SDK package enhances the L1 contract interaction and adds support for the new L1 claiming system, which is based on a Merkle tree and requires a Merkle proof for claiming.
```
Comment (jlwllmr) [inline docs/build-on-linea/linea-sdk/index.md:33]:
```suggestion
The updated Linea SDK package enhances the L1 contract interaction and adds support for the new L1 claiming system, which is based on a Merkle tree and requires a Merkle proof for claiming.
```

File: docs/build-on-linea/linea-sdk/index.md [hunk 7]
```diff
@@ -27,8 +27,45 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
...
+ 
+ There are three important things to note:
+  - The previous L1 claiming and all functions associated are still supported.
```
Comment (jlwllmr) [inline docs/build-on-linea/linea-sdk/index.md:36]: If it's still supported, wouldn't it be better described as "existing" rather than "previous"? 

Also, I think we're missing a noun after "claiming": claiming process, claiming flow, claiming mechanism, etc.

File: docs/build-on-linea/linea-sdk/index.md [hunk 8]
```diff
@@ -27,8 +27,45 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
...
+  - The L2 claiming remains unaltered, and all SDK features to interact with L2 will remain unchanged.
```
Comment (jlwllmr) [inline docs/build-on-linea/linea-sdk/index.md:37]:
```suggestion
  - The L2 claiming remains unaltered, and all SDK features for interacting with L2 will remain unchanged.
```

File: docs/build-on-linea/linea-sdk/index.md [hunk 9]
```diff
@@ -27,8 +27,45 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
...
+  - The previous L1 claiming function and code samples provided here cater to the transition period where pre-transition messages are claimed without the Merkle proof and post-transition with proof. If this SDK is being used after the transition, using the logic that switches between Merkle and non-Merkle proof claiming is sub-optimal.
```
Comment (jlwllmr) [inline docs/build-on-linea/linea-sdk/index.md:38]: I think this whole bullet needs some work for clarity. It's quite hard to follow. I don't have any concrete suggestions as I'm not entirely sure what it's trying to say, but I think the use of "pre-transition" and "post-transition" is potentially adding complexity. Let me know if you want me to take a second look and devote more time to this.

File: docs/build-on-linea/linea-sdk/index.md [hunk 10]
```diff
@@ -27,8 +27,45 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
...
+
+ The updated SDK introduces several new features for L1 interactions:
```
Comment (jlwllmr) [inline docs/build-on-linea/linea-sdk/index.md:40]:
```suggestion
The updated SDK introduces several new features for L1 interactions:
```

File: docs/build-on-linea/linea-sdk/index.md [hunk 11]
```diff
@@ -27,8 +27,45 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
...
+
+ A new L1ClaimingService class that includes the following functions:
```
Comment (jlwllmr) [inline docs/build-on-linea/linea-sdk/index.md:42]:
```suggestion
A new L1ClaimingService class that includes the following functions:
```

File: docs/build-on-linea/linea-sdk/index.md [hunk 12]
```diff
@@ -27,8 +27,45 @@ The SDK focuses on interacting with smart contracts on both Ethereum and Linea n
...
+  - getMessageProof: This function retrieves the message Merkle tree proof required for new message claims on L1.
+  - isClaimingNeedingProof: This function determines whether a proof is needed to claim a message.
+  - getMessageStatus: This function retrieves a message's status, returning the status of both old and new messages.
+  - estimateClaimMessageGas: This function provides an estimate of the gas cost for both old and new claim transactions.
+  - claimMessage: This function enables a message to be claimed using either the old or new function.
+
+ Two new functions in the L1 contract:
```
Comment (jlwllmr) [inline docs/build-on-linea/linea-sdk/index.md:49]:
```suggestion
Two new functions in the L1 contract:
```

**Review by jlwllmr — CHANGES_REQUESTED** [review]

**Review by jlwllmr — APPROVED** [review]
